osidb/models.py

```diff
@@ -1487,6 +1489,139 @@ def bzsync(self, *args, bz_api_key, **kwargs):
         self.flaw.save(*args, bz_api_key=bz_api_key, **kwargs)
 
 
+class CVSSManager(
+    ACLMixinManager,
+    TrackingMixinManager,
+):
+    """CVSS manager"""
+
+    @staticmethod
+    def create_cvss(flaw, affect, vector, issuer, version, **extra_fields):
+        """return a new CVSS or update an existing CVSS without saving"""
+
+        try:
+            cvss = CVSS.objects.get(
+                flaw=flaw,
+                affect=affect,
+                vector=vector,
+                issuer=issuer,
+                version=version,
+            )
+            return cvss
+
+        except ObjectDoesNotExist:
+            return CVSS(
+                flaw=flaw,
+                affect=affect,
+                vector=vector,
+                issuer=issuer,
+                version=version,
+                **extra_fields,
+            )
+
+
+class CVSS(
+    ACLMixin,
+    AlertMixin,
+    TrackingMixin,
+):
+    """Model CVSS"""
+
+    class Meta:
+        """define meta"""
+
+        constraints = [
+            models.UniqueConstraint(
+                fields=["flaw", "version", "issuer"], name="unique CVSS of a Flaw"
+            ),
+            models.UniqueConstraint(
+                fields=["affect", "version", "issuer"], name="unique CVSS of an Affect"
+            ),
+        ]
+
+    class CVSSVersion(models.TextChoices):
+        VERSION2 = "V2", "version 2"
+        VERSION3 = "V3", "version 3"
+
+    class CVSSIssuer(models.TextChoices):
+        REDHAT = "RH", "Red Hat"
+        NIST = "NIST", "NIST"
+
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    flaw = models.ForeignKey(
+        Flaw, on_delete=models.CASCADE, blank=True, related_name="cvss_scores"
+    )
+
+    affect = models.ForeignKey(
+        Affect,
+        on_delete=models.CASCADE,
+        blank=True,
+        null=True,
+        related_name="cvss_scores",
+    )
```

Comment:
I would deal with the relations in the child classes. The reason is that a single CVSS record cannot relate to a flaw and an affect at the same time. It is always one or the other. I would simply create a child classes `FlawCVSS` and `AffectCVSS` which would account for the relations so those relations would be removed from the base `CVSS`.